walletdb/interface.go

```diff
@@ -207,23 +208,60 @@ type DB interface {
 	Close() error
 }
 
-// BatchDB is a special version of the main DB interface that allos the caller
-// to specify write transactions that should be combine dtoegether if multiple
+// BatchDB is a special version of the main DB interface that allows the caller
+// to specify write transactions that should be combined toegether if multiple
 // goroutines are calling the Batch method.
 type BatchDB interface {
 	DB
 
 	// Batch is similar to the package-level Update method, but it will
-	// attempt to optismitcally combine the invocation of several
+	// attempt to optimistically combine the invocation of several
 	// transaction functions into a single db write transaction.
 	Batch(func(tx ReadWriteTx) error) error
 }
 
+// ExtendedDB is a special version of the main DB interface that provides their
+// own implementation of the View and Update methods. The walletdb package will
+// use those methods instead of the package global View/Update functions if the
+// given DB is compatible with this extended interface.
+type ExtendedDB interface {
```

Comment:
I'm thinking if it is feasible to instead of introducing the `ExtendedDB` interface, we could just change the `DB` interface to include `View` and `Update` with reset and we could keep the old manual txn based `walletdb.View` and `walletdb.Update` intact to simplify upgrades for other implementors? Then internally in the wallet we could just use the new extended `DB`. wdyt?

Reply:
Good idea, simplifies quite a lot! I didn't go ahead and rename/change anything in the larger code base. But we can now do so package by package if we wish to.